ogcapi-stable/ogcapi-features-core/src/main/java/de/ii/ogcapi/features/core/domain/SchemaDeriverJsonSchema.java

```diff
@@ -314,7 +317,10 @@ protected JsonSchema withConstraints(
             property.isArray()
                 ? property.getValueType().orElse(SchemaBase.Type.UNKNOWN) != SchemaBase.Type.INTEGER
                 : property.getType() != SchemaBase.Type.INTEGER;
-        Set<String> values = codelist.get().getData().getEntries().keySet();
+        Collection<String> values =
+            useCodelistKeys
+                ? codelist.get().getData().getEntries().keySet()
+                : codelist.get().getData().getEntries().values();
```

Comment:
The use of `values()` can lead to exceptions. This is a follow-on effect from my other comment. If there is an integer key, `string` will be false, but the `values` are proper strings and the `Integer.valueOf(val)` below will throw a NumberFormat exception.

Example case: https://demo.ldproxy.net/daraa/collections/TransportationGroundCrv/schemas/feature

Reply:
Well, the use of `keySet()` can also lead to exceptions, so this was unsafe code before the PR. I pushed a change that filters out non-integers.